### PR TITLE
FUTURE: override byonm with nodeModulesDir setting

### PR DESCRIPTION
**Pull request by dsherret**
Makes the `"nodeModulesDir"` setting take precedence over byonm when using `DENO_FUTURE`.

### CODE REVIEW

File: cli/args/mod.rs [hunk 1]
```diff
@@ -1595,10 +1591,13 @@ impl CliOptions {
   }
 
   pub fn use_byonm(&self) -> bool {
-    self.enable_future_features()
-  }
+    if self.enable_future_features()
+      && self.node_modules_dir_enablement().is_none()
+    {
+      return true;
+    }
```
Comment (bartlomieju) [inline cli/args/mod.rs:1598]: I don't think this is correct. This will force BYON even if you don't have `node_modules/` dir. That seems unwanted

Comment (dsherret) [reply cli/args/mod.rs:1598]: Oh wait, yeah it's not correct because it's not checking if the package.json exists.

Comment (dsherret) [reply cli/args/mod.rs]: Fixed to check if package.json exists.

**Review by nayeemrmn — APPROVED** [review]
LGTM

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**